executor/logging.go

```diff
@@ -2,23 +2,129 @@ package executor
 
 import (
 	"bufio"
+	"fmt"
 	"io"
 	"log"
+	"time"
 )
 
+var stdLogBufferSize = 64 * 1024
+
 // bindLoggingPipe spawns a goroutine for passing through logging of the given output pipe.
+//
+// This implementation is adapted from https://github.com/hashicorp/go-plugin/pull/98
 func bindLoggingPipe(name string, pipe io.Reader, output io.Writer) {
 	log.Printf("Started logging %s from function.", name)
 
-	scanner := bufio.NewScanner(pipe)
-	logger := log.New(output, log.Prefix(), log.Flags())
+	scanner := bufio.NewReaderSize(pipe, stdLogBufferSize)
 
 	go func() {
-		for scanner.Scan() {
-			logger.Printf("%s: %s", name, scanner.Text())
-		}
-		if err := scanner.Err(); err != nil {
-			log.Printf("Error scanning %s: %s", name, err.Error())
+		logFlags := log.Flags()
+		continuation := false
+
+		for {
+			line, isPrefix, err := scanner.ReadLine()
+			switch {
+			case err == io.EOF:
+				return
+			case err != nil:
+				log.Printf("Error scanning %s: %s", name, err.Error())
+				return
+			}
+
+			fmt.Println(isPrefix, string(line))
+
+			if !continuation {
+				// we are not continuing a previous line, so we
+				// start the log line with the log prefix
+				output.Write(logHeader(logFlags))
+			}
+
+			output.Write(line)
+			// the line is longer than stdLogBufferSize
+			if isPrefix || continuation {
+
+				// we are continuing a long log line,
+				// but it is the end of the actual line,
+				// so put the newline back in
+				if !isPrefix {
+					output.Write([]byte{'\n'})
+				}
+
+				continuation = isPrefix
+				continue
+			}
+
+			// not a prefix or continuation, so we have written a complete line
+			// so put the newline back in
+			output.Write([]byte{'\n'})
 		}
 	}()
 }
+
+// logHeader copies timestamp construction from the log package `formatHeader`
```

Comment:
Could you link to the source file please in the comment?

Reply:
it is documented now